openupgrade_scripts/scripts/project/14.0.1.1/post-migration.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2021 ForgeFlow S.L.  <https://www.forgeflow.com>
+# Copyright 2021 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def map_project_project_rating_status(env):
+    openupgrade.map_values(
+        env.cr,
+        openupgrade.get_legacy_name("rating_status"),
+        "rating_status",
+        [("no", "stage")],
+        table="project_project",
+    )
+
+
+def _fill_res_users_m2m_tables(env):
+    # TODO: Take into account channels and task followers part of the old rule
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO project_allowed_internal_users_rel
+        (project_project_id, res_users_id)
+        SELECT pp.id, ru.id
+        FROM project_project pp
+        JOIN mail_followers mf ON mf.res_model = 'project.project' AND mf.res_id = pp.id
+        JOIN res_users ru ON ru.partner_id = mf.partner_id AND NOT ru.share
+        WHERE pp.privacy_visibility = 'followers'
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO project_allowed_portal_users_rel
+        (project_project_id, res_users_id)
+        SELECT pp.id, ru.id
+        FROM project_project pp
+        JOIN mail_followers mf ON mf.res_model = 'project.project' AND mf.res_id = pp.id
+        JOIN res_partner rp ON mf.partner_id = rp.id
+            OR mf.partner_id = rp.commercial_partner_id
+        JOIN res_users ru ON ru.partner_id = rp.id AND ru.share
+        WHERE pp.privacy_visibility = 'portal'
+        """,
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        INSERT INTO project_task_res_users_rel
+        (project_task_id, res_users_id)
+        SELECT pt.id, rel.res_users_id
+        FROM project_allowed_internal_users_rel rel
+        JOIN project_task pt WHERE pt.project_id = rel.project_project_id
+        UNION
+        SELECT pt.id, rel.res_users_id
+        FROM project_allowed_portal_users_rel rel
+        JOIN project_task pt WHERE pt.project_id = rel.project_project_id
```

Comment:
```suggestion
        JOIN project_task pt ON pt.project_id = rel.project_project_id
        UNION
        SELECT pt.id, rel.res_users_id
        FROM project_allowed_portal_users_rel rel
        JOIN project_task pt ON pt.project_id = rel.project_project_id
```

Reply:
As I'm doing this between training sessions, I haven't put enough effort on checking SQLs. Sorry for the hammering.